packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -6957,7 +6957,7 @@ function buildFakeClient(
       rpcUrl: 'https://test.network',
     },
     provider,
-    blockTracker: new FakeBlockTracker(),
+    blockTracker: new FakeBlockTracker() as BlockTracker,
```

Comment:
@MajorLift This is an example of us needing to use a type assertion. I guess the type of `FakeBlockTracker` isn't quite right here and doesn't match the type of `blockTracker` that `NetworkClient` demands.

Reply:
@mcmire This seems like it's either been fixed or wasn't needed in the first place: https://github.com/MetaMask/core/pull/1833. These silent false negatives are the worse thing about type assertions IMO. 

In general, when encountering "not assignable to" errors, I would like the underlying types to be reconciled/aligned, at least eventually. Are there maybe "usual suspect" entities in our codebase that I can assume it'll be safe to use type assertion on?